transform/convert_quadkey_and_Vertical_id.go

```diff
@@ -576,6 +664,97 @@ func convertVerticallIDToBit(vZoom int64, vIndex int64, outputZoom int64, maxHei
+
+	var (
+		outputIndexes []int64
+		error         error
+	)
+
+	// determine the upper and lower index bounds to search for matches in height solution space
+	lowerBound, error := calculateMinVerticalIndex(inputIndex, inputZoom, outputZoom, altitudeRangeScalar, indexOffset)
+	if error != nil {
+		return nil, error
+	}
+	upperBound, error := calculateMinVerticalIndex(inputIndex+1, inputZoom, outputZoom, altitudeRangeScalar, indexOffset)
+	if error != nil {
+		return nil, error
+	}
+
+	// Determine the vertical index/indices to return.
+	// a) always return the lowerBound index. Regardless of the difference between the inputZoom and outputZoom,
+	// mathematically the altitude associated with the lower bounds will always satisfy the solution set.
+	// b) cycle through indices from lowerBounds+1 to upperBounds with i to find any possible additional indexes
+	// that satisfy the solution set.
+	outputIndexes = append(outputIndexes, lowerBound)
+
+	for i := lowerBound + 1; i < upperBound; i++ {
+
+		outputIndexes = append(outputIndexes, int64(i))
+	}
+
+	return outputIndexes, nil
+
+}
+
+// converts a vertical index from one set of zoom parameters to another disregarding the floor() cacluation. This creates a simplier system of equations where the solu	tion set for height is a single variable. However, this does not describe the full solution set of height since we have excluded the floor calculation; it describes the condition where m = x, given m = floor(x) if and only if m <= x < m +1;
+func calculateMinVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, altitudeRangeScalar int64, indexOffset int64) (int64, error) {
+
+	// check to make sure the input index exists in the input world
+	inputResolution := CalculateArithmeticShift(1, inputZoom)
+
+	maxInputIndex := int64(inputResolution - 1)
+	minInputIndex := -int64(inputResolution + 1)
+
+	outputResolution := CalculateArithmeticShift(1, outputZoom)
+	maxOutputIndex := int64(outputResolution-1) + indexOffset
+	minOutputIndex := -int64(outputResolution+1) + indexOffset
+
+	if inputIndex > maxInputIndex || inputIndex < minInputIndex {
+		return 0, errors.NewSpatialIdError(errors.InputValueErrorCode, "input index does not exist")
+	}
+
+	// note that in the case of decimals, int64 rounds down to the nearest integer. This is desired behavior.
+	outputIndex := (indexOffset) + CalculateArithmeticShift(inputIndex, (outputZoom-inputZoom+altitudeRangeScalar))
+
+	if outputIndex > maxOutputIndex || outputIndex < minOutputIndex {
+		return 0, errors.NewSpatialIdError(errors.InputValueErrorCode, "output index does not exist with given outputZoom, altitudeRangeScalar, and indexOffset")
+	}
+
+	return outputIndex, nil
+
+}
+
+// computes arithmatic shift of index and shift parameters. When index = 1, similar to returning 2^shift. When index > 1,
+// similar to returning index*2^shift.
+func CalculateArithmeticShift(index int64, shift int64) int64 {
```

Comment:
Why did you make only this function public?
You can test private functions.
If it is not necessary, revert it for reducing the maintenance costs and the learning costs.